Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -1011,7 +1013,13 @@ RefPtr<StyleRuleScope> CSSParserImpl::consumeScopeRule(CSSParserTokenRange prelu
         m_observerWrapper->observer().endRuleBody(m_observerWrapper->endOffset(block));
     }
 
-    auto rules = consumeRegularRuleList(block);
+    NestingLevelIncrementer incrementer { m_scopeRuleNestingLevel };
+    m_ancestorStack.append(AncestorRuleType::Scope);
```

Comment:
Maybe we could get rid of the scope/stylerule nesting level counters completely and just traverse the m_ancestorStack (which is usually empty or small).  No one seems to care about the actual count, just if there are any.

Reply:
Yes, I've opened a bugzilla to refactor those later https://bugs.webkit.org/show_bug.cgi?id=265566